report/renderer/details-renderer.js

```diff
@@ -64,8 +64,8 @@ export class DetailsRenderer {
         return null;
 
       default: {
-        // @ts-expect-error tsc thinks this is unreachable, but be forward compatible
```

Comment:
Can you split out the changes to the report code / the filtering for another PR? Just the changes in core alone are more than enough of a review chunk :)

Reply:
This can be done if it makes reviewing easier. Do you want to retain changes in `util.js` or fully disconnect `report/*`?

Reply:
Done: #14697